src/tslint-cli.ts

```diff
@@ -207,6 +207,22 @@ const processFile = (file: string) => {
         process.exit(1);
     }
 
+    const buffer = new Buffer(256);
+    buffer.fill(0);
+    const fd = fs.openSync(file, "r");
+    try {
+        fs.readSync(fd, buffer, 0, 256, null);
+        if (buffer.readInt8(0) === 0x47 && buffer.readInt8(188) === 0x47) {
+            // MPEG transport streams use the '.ts' file extension. They use 0x47 as the frame
+            // separator, repeating every 188 bytes. It is unlikely to find that pattern in
+            // TypeScript source, so tslint ignores files with the specific pattern.
+            console.warn(`${file}: ignoring MPEG transport stream`);
+            return;
+        }
+    } finally {
+        fs.closeSync(fd);
+    }
+
     const contents = fs.readFileSync(file, "utf8");
```

Comment:
I considered that, but another problem with videos is that they are often huge, hundreds of megabytes, which can be enough to kill the NodeJS process. So this code on purpose only peeks 256 bytes to make that decision.

Overall the biggest cost in reading the file is the hard disk, so after peeking it will be cached by the OS, and then parsing, which is unconcerned.

We could change this to keep reading from the file descriptor and appending into buffer, then converting that, but I don't think it'll be worth it.